src/typing/statement.ml

```diff
@@ -68,6 +68,11 @@ let function_desc ~async ~generator desc =
   | false, true -> spf "generator %s" desc
   | false, false -> desc
 
+let reject_explicit_this cx msg = function
```

Comment:
I don't know whether `this`-pseudoparams just work on objects.  There's a single test case where I move a method from the class's prototype to a compatible object, and it just works.  The tests should be extended to validate behavior on objects, or `this`-pseudoparams should be blacklisted on objects.